cmd/ctr/commands/run/run.go

```diff
@@ -163,7 +168,7 @@ var Command = cli.Command{
 		if err != nil {
 			return err
 		}
-		if context.Bool("rm") && !detach {
+		if rm && !detach {
```

Comment:
If we supplied rm we should know detach is false as we'd exit on line 159 now

Reply:
Although the upper logic provides some checks, I think the prerequisite for perform `defer container.Delete(ctx, containerd.WithSnapshotCleanup)` is strict `rm && !detach`. 
For robustness I did not remove the judgment on `detach` here, what do you think?

Of course, for the sake of code brevity, I think it's fine to just judge x

Reply:
It was mainly a nit, so I'm fine with this!